app/actions.py

```diff
@@ -92,30 +92,170 @@ def performDeleteRange(self, upperRow, upperCol, lowerRow, lowerCol):
         self.getText(upperRow, upperCol, lowerRow, lowerCol)))
       self.redo()
 
+  def dataToBookmark(self):
+    """
+    Grabs all the cursor data and returns a bookmark.
+
+    Args:
+      None.
+
+    Returns:
+      A bookmark in the form of (bookmarkRange, bookmarkData).
+      bookmarkRange is an ordered tuple in which its elements
+      are the rows that the bookmark affects.
+      bookmarkData is a dictionary that contains the cursor data.
+    """
+    cursor = (self.view.cursorRow, self.view.cursorCol)
+    pen = (self.penRow, self.penCol)
+    marker = (self.markerRow, self.markerCol)
+    selectionMode = self.selectionMode
+    bookmarkData = {
+      'cursor': cursor,
+      'marker': marker,
+      'pen': pen,
+      'selectionMode': selectionMode,
```

Comment:
I like use temps when the lines are too long. In this case, it looks like the values will fit nicely (i.e. readably) directly in lines 113 to 116. WDYT?

Reply:
Sorry, I'm not really understanding what you mean. What do you mean by using temps and what values are you suggesting to inject?

Reply:
Something like this
```
  bookmarkData = {
    'cursor': (self.view.cursorRow, self.view.cursorCol),
    'marker': (self.markerRow, self.markerCol),
    'pen': (self.penRow, self.penCol),
    'selectionMode': self.selectionMode,
  }
```

Reply:
Ahh I see. Done.